bundle/config/mutator/apply_presets.go

```diff
@@ -221,6 +222,19 @@ func (m *applyPresets) Apply(ctx context.Context, b *bundle.Bundle) diag.Diagnos
 		dashboard.DisplayName = prefix + dashboard.DisplayName
 	}
 
+	if config.IsExplicitlyEnabled((b.Config.Presets.InPlaceDeployment)) {
+		root := b.SyncRoot.Native()
+		_, ok := env.Lookup(ctx, envDatabricksRuntimeVersion)
+		isInWorkspace := ok && strings.HasPrefix(root, "/Workspace/")
+
+		if isInWorkspace {
+			b.Config.Workspace.FilePath = b.BundleRootPath
```

Comment:
Do we also need to do something with `resource_path` and `artifact_path`?

For `resource_path` we should make sure that dashboards (like referred in a job maybe) works with in place deployments.

For `artifact_path`, maybe we can avoid uploading artifacts that are built and directly refer the local path to the `whl` / `jar` file?